projects/packages/jetpack-mu-wpcom/src/features/launchpad/class-launchpad-task-lists.php

```diff
@@ -373,11 +373,26 @@ public function add_hooks_for_active_tasks( $task_list_id = null ) {
 			return;
 		}
 
-		$active_tasks = $this->get_active_tasks( $task_list_id );
-		foreach ( $active_tasks as $task ) {
-			$task_definition = $this->get_task( $task['id'] );
+		$task_list_id = $task_list_id ? $task_list_id : get_option( 'site_intent' );
+		// Sites without a `site_intent` option will not have any tasks.
+		if ( ! $task_list_id ) {
+			return;
+		}
+
+		$task_list = $this->get_task_list( $task_list_id );
+		if ( empty( $task_list ) || ! isset( $task_list['task_ids'] ) ) {
+			return;
+		}
+
+		foreach ( $task_list['task_ids'] as $task_id ) {
+			$task_definition = $this->get_task( $task_id );
 			if ( isset( $task_definition['add_listener_callback'] ) && is_callable( $task_definition['add_listener_callback'] ) ) {
-				call_user_func_array( $task_definition['add_listener_callback'], array( $task, $task_definition ) );
+				// We only need to know the built completion status if the task has an `add_listener_callback` property.
+				// Small optimization to not run `is_complete_callback` as often.
+				$task = $this->build_task( $task_definition );
+				if ( ! $task['completed'] ) {
```

Comment:
Let's add an `is_callable( $task_definition['add_listener_callback'] )` check to prevent errors if there's no callback

Reply:
this https://github.com/Automattic/jetpack/pull/30527/files#diff-372d5b25a712801a764dd8dbf52fe2362df78fb2a91a015f3d7f2225bd65f0ebR389 ?

Reply:
Ha! I did not see that. Thanks

Reply:
@obenland good catch thanks! 860d8cd213520e363e1699b2b35ebfa4f3ee9f57

Reply:
Oh wait lol I just caught it now too haha

Reply:
Maybe we can move the check closer to where the callback gets called?

Reply:
I wanted the check the far prior because if it doesn't pass then I don't have to build the task.